Sources/UI/RequestDetailViewController.swift

```diff
@@ -20,44 +20,56 @@ class RequestDetailViewController: WHBaseViewController {
         Section(name: "Response Header", type: .responseHeader),
         Section(name: "Response Body", type: .responseBody)
     ]
-    
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        
+
         if let urlString = request?.url{
             title = URL(string: urlString)?.path
         }
-        
+
         let shareButton = UIBarButtonItem(barButtonSystemItem: .action, target: self, action: #selector(shareContent))
         navigationItem.rightBarButtonItems = [shareButton]
-        
+
         tableView.estimatedRowHeight = 100.0
         tableView.rowHeight = UITableView.automaticDimension
         tableView.register(UINib(nibName: "TextTableViewCell", bundle:WHBundle.getBundle()), forCellReuseIdentifier: "TextTableViewCell")
         tableView.register(UINib(nibName: "ActionableTableViewCell", bundle:WHBundle.getBundle()), forCellReuseIdentifier: "ActionableTableViewCell")
         tableView.register(UINib(nibName: "RequestTitleSectionView", bundle:WHBundle.getBundle()), forHeaderFooterViewReuseIdentifier: "RequestTitleSectionView")
     }
-    
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-    }
-    
+
     @objc func shareContent(){
         if let request = request{
-            let textShare = [RequestModelBeautifier.txtExport(request: request)]
-            let customItem = CustomActivity(title: "Save to the desktop", image: UIImage(named: "activity_icon", in: WHBundle.getBundle(), compatibleWith: nil)) { (sharedItems) in
-                guard let sharedStrings = sharedItems as? [String] else { return }
-                
-                for string in sharedStrings {
-                    FileHandler.writeTxtFileOnDesktop(text: string, fileName: "\(Int(Date().timeIntervalSince1970))-wormholy.txt")
-                }
+
+            let alert = UIAlertController(title: "Export", message: "", preferredStyle: .actionSheet)
+            let action = UIAlertAction(title: "cURL", style: .default) { [weak self] item in
+                self?.export(share: RequestModelBeautifier.curlExport(request: request))
+            }
+
+            let action2 = UIAlertAction(title: "Text dump", style: .default) { [weak self] item in
+                self?.export(share: RequestModelBeautifier.txtExport(request: request))
+            }
+
+            alert.addAction(action)
+            alert.addAction(action2)
+            self.present(alert, animated: true, completion: nil)
+        }
+    }
+
+    func export(share payload: String) {
+        let customItem = CustomActivity(title: "Save to desktop", image: UIImage(named: "activity_icon", in: WHBundle.getBundle(), compatibleWith: nil)) { (sharedItems) in
+            guard let sharedStrings = sharedItems as? [String] else { return }
+
+            for string in sharedStrings {
+                FileHandler.writeTxtFileOnDesktop(text: string, fileName: "\(Int(Date().timeIntervalSince1970))-wormholy.txt")
```

Comment:
will it work on device? currently it's kind of painful to share requests from simulator or from device, I have to export in notes and make them use the same icloud account which feels so wrong...

Reply:
It should, the same way it already exported the contents